api/krusty/remoteloader_test.go

```diff
@@ -0,0 +1,357 @@
+    name: nginx
+`
+	var multibaseDevExampleBuild = strings.ReplaceAll(simpleBuild, "myapp-pod", "dev-myapp-pod")
+
+	repos := createGitRepos(t)
+	tests := []struct {
+		name          string
+		kustomization string
+		expected      string
+		err           string
+		skip          bool
+	}{
+		{
+			name: "simple",
+			kustomization: `
+resources:
+- file://$ROOT/simple.git
+`,
+			expected: simpleBuild,
+		},
+		{
+			name: "has path",
+			kustomization: `
+resources:
+- file://$ROOT/multibase.git/dev
+`,
+			expected: multibaseDevExampleBuild,
+		},
+		{
+			name: "has ref",
+			kustomization: `
+resources: 
+- "file://$ROOT/simple.git?ref=change-image"
+`,
+
+			expected: strings.ReplaceAll(simpleBuild, "nginx:1.7.9", "nginx:2"),
+		},
+		{
+			// Version is the same as ref
+			name: "has version",
+			kustomization: `
+resources:
+- file://$ROOT/simple.git?version=change-image
+`,
+			expected: strings.ReplaceAll(simpleBuild, "nginx:1.7.9", "nginx:2"),
+		},
+		{
+			name: "has submodule",
+			kustomization: `
+resources:
+- file://$ROOT/with-submodule.git/submodule
+`,
+			expected: simpleBuild,
+		},
+		{
+			name: "has submodule but not initialized",
+			kustomization: `
+resources:
+- file://$ROOT/with-submodule.git/submodule?submodules=0
+`,
+			err: "unable to find",
+		},
+		{
+			name: "has origin annotation",
+			skip: true, // The annotated path should be "pod.yaml" but is "notCloned/pod.yaml"
+			kustomization: `
+resources:
+- file://$ROOT/simple.git
+buildMetadata: [originAnnotations]
+`,
+			expected: `apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    config.kubernetes.io/origin: |
+      path: pod.yaml
+      repo: file://$ROOT/simple.git
+  labels:
+    app: myapp
+  name: myapp-pod
+spec:
+  containers:
+  - image: nginx:1.7.9
+    name: nginx
+`,
+		},
+		{
+			name: "has ref path and origin annotation",
+			kustomization: `
+resources:
+- file://$ROOT/multibase.git/dev?version=main
+buildMetadata: [originAnnotations]
+`,
+			expected: `apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    config.kubernetes.io/origin: |
+      path: base/pod.yaml
+      repo: file://$ROOT/multibase.git
+      ref: main
+  labels:
+    app: myapp
+  name: dev-myapp-pod
+spec:
+  containers:
+  - image: nginx:1.7.9
+    name: nginx
+`,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.skip {
+				t.SkipNow()
+			}
+
+			kust := strings.ReplaceAll(test.kustomization, "$ROOT", repos.root)
+			fSys, tmpDir := createKustDir(t, kust)
+
+			b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+			m, err := b.Run(
+				fSys,
+				tmpDir.String())
+
+			if test.err != "" {
+				assert.Error(t, err)
+				if err != nil {
+					assert.Contains(t, err.Error(), test.err)
+				}
+			} else {
+				assert.NoError(t, err)
+				if err == nil {
+					checkYaml(t, m, strings.ReplaceAll(test.expected, "$ROOT", repos.root))
+				}
+			}
+		})
+	}
+}
+
+func TestRemoteLoad_RemoteProtocols(t *testing.T) {
```

Comment:
Please add error test cases both here and for the file protocol showing what happens when the URL is bad (coverage formerly in TestRemoteResourceWithHttpError).